Assets/Scripts/UI/TextBox.cs

```diff
@@ -0,0 +1,70 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.UI;
+
+public class TextBox : MonoBehaviour {
+
+	Text _UIText = null;
+
+	[SerializeField] KeyCode _clearKey = 0;
+
+	Coroutine _textFade;
+
+	void Start ()
+	{
+		_UIText = GetComponent<Text>();
+	}
+
+	void Update()
+	{
+		if ( Input.GetKeyDown( _clearKey ) )
+		{
+			ClearText();
+			CancelInvoke( "ClearText" );
```

Comment:
Where is this being Invoked that it needs to be cancelled? I would expect that to be in a public function on this object.